fix(CI): remove strict check

maqi opened the pull request:
There is node churning during the memcheck test, which does have a chance there will be failed fetch alarm against dead node.

### Description

Please provide a brief description of the changes made in this pull request. Highlight the purpose of the changes and the problem they address.

### Related Issue

Fixes #<issue_number> (if applicable).

### Type of Change

Please mark the types of changes made in this pull request.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please describe):

### Checklist

Please ensure all of the following tasks have been completed:

- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have updated the documentation accordingly.
- [x] I have followed the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines for commit messages.
- [x] I have verified my commit messages with [commitlint](https://commitlint.js.org/#/).

<!--
### Footer (Hidden from GitHub view)
This template uses [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). Please ensure your commit messages follow these guidelines for clarity and consistency.

Commit messages should be verified using [commitlint](https://commitlint.js.org/#/).

Commit Message Format:
<type>[optional scope]: <description>
[optional body]
[optional footer(s)]

Common Types:
- `feat`
- `fix`
- `docs`
- `style`
- `refactor`
- `perf`
- `test`
- `build`
- `ci`
- `chore`
- `revert`
-->

b-zee commented on sn_node/src/node/messaging/mod.rs:
```diff
@@ -78,10 +81,11 @@ impl MyNode {
                 // Anythign returned here means there's an issue and we should
                 // short-circuit below
-                let ae_cmds = self
+                let ae_cmds = read_locked_node
```
```suggestion
                let ae_cmds = node.read().await
```

This doesn't make a difference currently, but in the future it might prevent keeping the lock just that little bit longer than needed.

b-zee reviewed (approved):
Looks a bit more messy in general, but looks like the right thing is done.

maqi committed:
fix(CI): remove strict check

There is node churning during the memcheck and location verification
tests, which does havea chance of failed fetch alarm against dead node.
Hence move the check only to be undertaken withn the large file test,
which all nodes are persistant.

mazzi reviewed (approved):
LGTM 🚀
